### PR TITLE
chore: support for manual read replica

### PR DESCRIPTION
**Pull request by rebelchris**
Support for 1 read replica slave.
Set the default to still use master, as we often want direct results on mutations from the master.

This way we can manually query the slave if we want it.
Example duplicates a endpoint for testing it: (modified data for testing purposes)

Screenshot 2024-09-09 at 16 46 20
Screenshot 2024-09-09 at 16 46 28

### CODE REVIEW

File: src/data-source.ts [hunk 1]
```diff
@@ -20,4 +13,25 @@ export const AppDataSource = new DataSource({
...
+        process.env.TYPEORM_DATABASE ||
+        (process.env.NODE_ENV === 'test' ? 'api_test' : 'api'),
+    },
+    slaves: [
+      {
+        host: process.env.TYPEORM_READ_HOST || 'localhost',
+        port: 5432,
+        username: process.env.TYPEORM_READ_USERNAME || 'postgres',
+        password: process.env.TYPEORM_READ_PASSWORD || '12345',
+        database: process.env.TYPEORM_READ_DATABASE,
```
Comment (omBratteng) [inline src/data-source.ts:33]: Fallback for adhoc?

Comment (rebelchris) [reply src/data-source.ts:33]: We can set fallback to test db? (we need some fallback else it will fail to spool.
Let me know what you think?

Comment (capJavert) [reply src/data-source.ts:33]: Just specify master as slave in adhoc (so we can still use `slave` in query build?

Comment (capJavert) [reply src/data-source.ts:33]: (to avoid having another db locally as you said)

Comment (rebelchris) [reply src/data-source.ts:33]: Yeah that also works :)

File: .infra/Pulumi.adhoc.yaml [hunk 2]
```diff
@@ -80,7 +80,7 @@ config:
     typeormDatabase: api
     typeormHost: postgres
     typeormPassword: 12345
-    typeormUser: postgres
+    typeormUsername: postgres
```
Comment (rebelchris) [inline .infra/Pulumi.adhoc.yaml:83]: Noticed this needed to be `username` 🫠

File: src/schema/sources.ts [hunk 3]
```diff
@@ -367,6 +367,66 @@ export const typeDefs = /* GraphQL */ `
       categoryId: String
     ): SourceConnection!
 
+    """
```
Comment (idoshamun) [inline src/schema/sources.ts:370]: That's for debugging right?

File: .infra/Pulumi.adhoc.yaml (hunk 2 above)
Comment (capJavert) [reply .infra/Pulumi.adhoc.yaml]: probably was defaulting to `postgres`?

File: src/schema/sources.ts (hunk 3 above)
Comment (rebelchris) [reply src/schema/sources.ts:370]: Yeah just a duplication but using the slave specifically :)

**Review by idoshamun — APPROVED** [review]

**Review by omBratteng — APPROVED** [review]

File: src/schema/sources.ts [hunk 4]
```diff
@@ -367,6 +367,66 @@ export const typeDefs = /* GraphQL */ `
...
+      """
+      Filter by category
+      """
+      categoryId: String
+    ): SourceConnection!
+
+    """
+    Get all available sources from read replica
+    """
+    sourcesReadReplica(
```
Comment (capJavert) [inline src/schema/sources.ts:403]: did you add double definition or is diff just funky?

Comment (rebelchris) [reply src/schema/sources.ts:403]: @capJavert Yeah it's just a duplicate of sources to verify on production.
Idea is to merge -> test via postman -> revert this change and make the main one use the RR.

Just wanted to be safe and not break prod on deployment if that makes sense?

Comment (capJavert) [reply src/schema/sources.ts:403]: no i get it but sourcesReadReplica query is defined twice?

Comment (rebelchris) [reply src/schema/sources.ts:403]: Oh sorry, missed that somehow.

**Review by capJavert — APPROVED** [review]

**Pull request merged**